core/src/test/scala/unit/kafka/admin/AdminRackAwareTest.scala

```diff
@@ -0,0 +1,198 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin
+
+import junit.framework.Assert._
+import kafka.common.TopicAndPartition
+import kafka.utils.{TestUtils, ZkUtils, Logging}
+import kafka.zk.ZooKeeperTestHarness
+import org.I0Itec.zkclient.ZkClient
+import org.junit.Test
+import org.scalatest.junit.JUnit3Suite
+
+import scala.collection.{mutable, Map, Seq}
+
+class AdminRackAwareTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
+
+  @Test
+  def testGetRackAlternatedBrokerList(): Unit = {
+    val rackMap = Map(0 -> "rack1", 1 -> "rack3", 2 -> "rack3", 3 -> "rack2", 4 -> "rack2", 5 -> "rack1")
+    val newList = AdminUtils.getRackAlternatedBrokerList(rackMap)
+    assertEquals(List(0, 3, 1, 5, 4, 2), newList)
+    val anotherList = AdminUtils.getRackAlternatedBrokerList(rackMap - 5)
+    assertEquals(List(0, 3, 1, 4, 2), anotherList)
+    val brokerList = List(0, 1, 2, 3, 4, 5)
+    val assignment: scala.collection.Map[Int, Seq[Int]] = AdminUtils.assignReplicasToBrokers(brokerList, 7, 3, 0, 0, rackMap)
+    val expected: Map[Int, Seq[Int]] = Map(0 -> List(0,3,1),
+                                           1 -> List(3,1,5),
+                                           2 -> List(1,5,4),
+                                           3 -> List(5,4,2),
+                                           4 -> List(4,2,0),
+                                           5 -> List(2,0,3),
+                                           6 -> List(0,4,2))
+    assertEquals(expected, assignment)
+  }
+
+  @Test
+  def testAssignmentWithRackAware() {
+    val brokerList = List(0, 1, 2, 3, 4, 5)
+    val brokerRackMapping = Map(0 -> "rack1", 1 -> "rack2", 2 -> "rack2", 3 -> "rack3", 4 -> "rack3", 5 -> "rack1")
+    val assignment: scala.collection.Map[Int, Seq[Int]] = AdminUtils.assignReplicasToBrokers(brokerList, 6, 3, 2, 0, brokerRackMapping)
+    ensureRackAwareAndEvenDistribution(assignment, brokerRackMapping, 6, 6, 3)
```

Comment:
It is a bug that by default the three verifications are disabled. I will fix it.